### PR TITLE
Use Unicode 13 for Character toUpperCase and toLowerCase

### PR DESCRIPTION
**Pull request by ekrich**
Matches JDK 15+ See - https://github.com/ekrich/scala-unicode/blob/master/src/main/scala/CaseUpperLowerJdk15Unicode13Test.scala

Run with AdoptOpenJDK 15 with no differences.

### CODE REVIEW

**Comment by sjrd** [comment]
If we're adapting some of the tables to a newer version of Unicode, I believe we should adapt all the Unicode-related tables at once. WDYT?

**Comment by ekrich** [comment]
I think that is a good idea. We would have to dust off the other inline scripts. Are you suggesting I take a look at the other tables?

**Comment by sjrd** [comment]
If you can have a look at the other tables, yes, please, that would be very welcome.

**Comment by LeeTibbert** [comment]
I also support a move to Unicode 13.  End users and application developers are going to care about
current accurate results, not adherence to a 7 year old standard.

@ekrich do you/we know which codepoints change their case attribute.  If yes, it would be a kind
thing to do for both Scala Native devos, and those using SN to develop if they could be release noted.

If any of the codepoints changed their case attribute, then the current PR will introduce skew with
regards to the scalanative regex code underlying javalib regex.  I quite distinctly remember
spending quite a bit of time and energy making the regex code bug-for-bug compatible with
the SN javalib Character code.  At that time, I believe that I was down-porting from something
like Unicode 7.0 to SN 6.??. 

PR #1685 is updated and shovel ready.  It changes the re2s regex used in SN to use the
javalib Character toMumble methods.  Eric & I agreed that this was a good way to keep
regex and Character consistent.  In addition, it removed some largish (and error prone)
tables from re2s regex.  This was done to also ease an almost certain upgrade of 
Unicode version, such as here.

I believe the .scala files in #1685 should work without change after the current PR is merged.
The corresponding Test files may fail, due to some of the Java 8 hand bug-for-bug compatibility
changes I mentioned above.

By the way, changing all the Unicode data tables of re2s will probably take someone longer than a long night.
Yes, much can be generated, but quirks abound, probably fewer with a up-to-date Unicode.

File: javalib/src/main/scala/java/lang/Character.scala [hunk 1]
```diff
@@ -1338,7 +1338,7 @@ object Character {
   }
 
   // Tables to support toUpperCase and toLowerCase transformations
-  // with Unicode 6.3.0 to match JDK8. This now uses UnicodeData.txt.
+  // with Unicode 13.0.0 to match JDK15 and JDK16(LTS).
```
Comment (sideeffffect) [inline javalib/src/main/scala/java/lang/Character.scala:1341]: JDK *17* will be LTS
https://en.wikipedia.org/wiki/Java_version_history

**Review by sjrd — APPROVED** [review]

**Pull request merged**